src/Resources/config/menu.xml

```diff
@@ -49,41 +38,32 @@
             <tag name="knp_menu.provider" />
         </service>
 
-        <service id="knp_menu.menu_provider.container_aware" class="%knp_menu.menu_provider.container_aware.class%" public="false">
-            <argument type="service" id="service_container" />
-            <argument type="collection" />
-        </service>
-
         <service id="knp_menu.menu_provider.builder_service" class="Knp\Bundle\MenuBundle\Provider\BuilderServiceProvider" public="false">
```

Comment:
this service must be removed. It is useless now that the lazy provider can always be used (see the compiler pass always removing it). And the class should be removed too.